ros/api/v1/hosts.py

```diff
@@ -376,57 +380,145 @@ def get(self, host_id):
+        "conditions": fields.Nested(condtions),
+        "meta": fields.Nested(meta)
+    }
+
+    @marshal_with(report_fields)
     def get(self):
+        account_number = identity(request)['identity']['account_number']
+        system_queryset = system_ids_by_account(account_number, fetch_records=True)
+
+        # System counts
+        all_systems_count = system_queryset.count()
+        optimized_systems = count_per_state(system_queryset, {'state': "Optimized"})
+        under_pressure_systems = count_per_state(system_queryset, {'state': "Under pressure"})
+        undersized_systems = count_per_state(system_queryset, {'state': "Undersized"})
+        oversized_systems = count_per_state(system_queryset, {'state': "Oversized"})
+        waiting_for_data_systems = count_per_state(system_queryset, {'state': "Waiting for data"})
+        idling_systems = count_per_state(system_queryset, {'state': "Idling"})
+
+        non_optimized = system_queryset.filter(
+            or_(
+                System.state == 'Oversized',
+                System.state == 'Undersized',
+                System.state == 'Idling',
+                System.state == 'Under pressure'
+            )
+        )
+        non_optimized_count = non_optimized.count()
+
+        # cpu counts
+        total_cpu_issues = sum([
+            len(system.cpu_states) if system.cpu_states else 0
+            for system in non_optimized.filter(System.cpu_states is not None)
+        ])
+        cpu_undersized = non_optimized.filter(System.cpu_states.any(SubStates.CPU_UNDERSIZED.value)).count()
+        cpu_under_pressure = non_optimized.filter(System.cpu_states.any(
+            SubStates.CPU_UNDER_PRESSURE.value)
+        ).count()
+        cpu_oversized = non_optimized.filter(System.cpu_states.any(SubStates.CPU_OVERSIZED.value)).count()
+
+        # memory counts
+        total_memory_issues = sum([
+            len(system.memory_states) if system.memory_states else 0
+            for system in non_optimized.filter(System.memory_states is not None)
+        ])
+        memory_undersized = non_optimized.filter(System.memory_states.any(SubStates.MEMORY_UNDERSIZED.value)).count()
+        memory_pressure = non_optimized.filter(System.memory_states.any(
+            SubStates.MEMORY_UNDER_PRESSURE.value)
+        ).count()
+        memory_oversized = non_optimized.filter(System.memory_states.any(SubStates.MEMORY_OVERSIZED.value)).count()
+
+        # io counts
+        total_io_issues = sum([
+            len(system.io_states) if system.io_states else 0
+            for system in non_optimized.filter(System.io_states is not None)
+        ])
+        io_undersized = non_optimized.filter(System.io_states.any(SubStates.IO_UNDERSIZED.value)).count()
+        io_pressure = non_optimized.filter(System.io_states.any(SubStates.IO_UNDER_PRESSURE.value)).count()
+        io_oversized = non_optimized.filter(System.io_states.any(SubStates.IO_OVERSIZED.value)).count()
```

Comment:
@saltgen, in my opinion, multiple filter queries and looping on those filtered systems is creating unnecessary complexity here. Can't you simple iterate on existing non_optimized systems ONLY once and calculate summation against each?